### PR TITLE
Fix horizontal scrolling regression (fixes #7404)

### LINKED ISSUE #7404: With touchpad horizontal scrolling is inverted
When I use the touchpad, scrolling upwards scrolls upward, downward scrolls downwards. But scrolling right scrolls to left.

1.16 and c9eae7c affected.
When using the OS' settings to activate `natural scrolling`, all four directions are inverted, so the bug remains.

Using Linux with KDE on top of X11 with libinput.
Other programs are not affected by this problem.
Scrolling by moving the mouse pointer via the touchpad to a side or corner of the screen works correct as well.

### PR DESCRIPTION
**Pull request by pehjota**
The old issue #2218 was actually a bug in SDL, fixed on X11 in version 2.0.18 and on Wayland in version 2.0.20.  The hardcoded workaround in pull #2481 (commit 4bc4373) caused a regression in fixed SDL versions.

This fix is similar to the workaround in widelands/widelands#5394 committed as widelands/widelands@67db32a.

Tested with SDL 2.0.14 and 2.28.5.

### CODE REVIEW

File: src/controller_base.cpp [hunk 1]
```diff
@@ -232,11 +232,32 @@ void controller_base::handle_event(const SDL_Event& event)
...
+		{
+			int xmul = -1;
+#if !defined(_WIN32) && !defined(__APPLE__)
+			const char* video_driver = SDL_GetCurrentVideoDriver();
+			SDL_version ver;
+			SDL_GetVersion(&ver);
+			if(video_driver != nullptr && ver.major <= 2 && ver.minor <= 0) {
+				if(std::strcmp(video_driver, "x11") == 0 && ver.minor < 18) {
+					xmul = 1;
+				} else if(std::strcmp(video_driver, "wayland") == 0 && ver.minor < 20) {
```
Comment (soliton-) [inline src/controller_base.cpp:254]: Shouldn't this and above check `ver.patch`?

Comment (pehjota) [reply src/controller_base.cpp]: Oops, yes, somehow I copied and pasted wrong from code I wrote elsewhere.  Fixed, thanks.

**Comment by soliton-** [comment]
This looks good to me. If someone can confirm that it works as expected that'd be cool but even if not I think we should just merge this.

**Comment by pehjota** [comment]
I see `mouse_handler_base::mouse_wheel()` calls virtual methods such as `mouse_wheel_left()` and `mouse_wheel_right()`, but the X-axis calls are incorrect.  It looks like the original intent in commit 821b009 was to follow SDL's coordinate signs (positive up and right), but this later changed to Wesnoth's map coordinate signs (positive down and right) in commit ea3bd56, and commit dfe2f33 reversed the X axis (positive down and left, opposite SDL) without updating the `mouse_wheel_*()` calls.  Although commit c912f7e removed the definitions of these methods, so this currently doesn't really matter.

I've pushed a second commit adding two more lines to the comment block, which should probably be squashed into the first commit.

And a third commit, which you can either merge or ignore, makes `mouse_handler_base::mouse_wheel()` use the same coordinate signs as the map itself and also fixes the `mouse_wheel_*()` virtual method calls.

**Comment by sevu** [comment]
Thank you!
Tested it and works as expected.

**Comment by Pentarctagon** [comment]
This can be merged then?

**Comment by Pentarctagon** [comment]
Thanks!

**Comment by pehjota** [comment]
Thanks for merging! Can/should this be cherry picked to 1.16?

**Comment by Pentarctagon** [comment]
Up to you if you think it's worth it, given 1.18 is planned to be released in March and the next (final) 1.16 release that would include this fix will be released at the same time.

**Comment by pehjota** [comment]
OK, I cherry picked, built, tested again with broken and fixed SDL versions, and sent a PR.  Too bad I missed 1.16.11 by just two days, but it'll be nice to at least get it into the final 1.16 release.

**Pull request merged**